server/services/v1/database/query_runner.go

```diff
@@ -179,7 +179,21 @@ func (runner *BaseQueryRunner) getDatabase(_ context.Context, tx transaction.Tx,
 	if tx != nil && tx.Context().GetStagedDatabase() != nil {
 		// this means that some DDL operation has modified the database object, then we need to perform all the operations
 		// on this staged database.
-		return tx.Context().GetStagedDatabase().(*metadata.Database), nil
+
+		db, ok := tx.Context().GetStagedDatabase().(*metadata.Database)
+		if !ok {
+			return nil, errors.Internal("invalid transaction staged database")
+		}
+
+		if db.Name() != projName {
+			return nil, errors.InvalidArgument("collections should belong to the same project in the transaction")
+		}
+
+		if db.BranchName() != branch && (branch != "" || db.BranchName() != metadata.MainBranch) {
```

Comment:
`if db.IsBranch() && db.BranchName() != branch {}`

Reply:
Could we move this check into a function like `db.IsBranch(metaData)` I think we use something similar in a few places.